shared/actions/gregor.js

```diff
@@ -119,21 +120,22 @@ function* handleTLFUpdate(items: Array<Types.NonNullGregorItem>): Saga.SagaGener
 
 function* handleBannersAndBadges(items: Array<Types.NonNullGregorItem>): Saga.SagaGenerator<any, any> {
   const sawChatBanner = items.find(i => i.item && i.item.category === 'sawChatBanner')
-  const sawSubteamsBanner = items.find(i => i.item && i.item.category === 'sawSubteamsBanner')
-  const chosenChannels = items.find(i => i.item && i.item.category === 'chosenChannelsForTeam')
-  const chosenChannelsForTeam =
-    (chosenChannels &&
-      chosenChannels.item &&
-      chosenChannels.item.body &&
-      chosenChannels.item.body.toString()) ||
-    JSON.stringify([])
   if (sawChatBanner) {
     yield Saga.put(TeamsGen.createSetTeamSawChatBanner())
   }
+
+  const sawSubteamsBanner = items.find(i => i.item && i.item.category === 'sawSubteamsBanner')
   if (sawSubteamsBanner) {
     yield Saga.put(TeamsGen.createSetTeamSawSubteamsBanner())
   }
-  yield Saga.put(TeamsGen.createSetChosenChannelsForTeam({chosenChannelsForTeam}))
+
+  const chosenChannels = items.find(i => i.item && i.item.category === 'chosenChannelsForTeam')
+  const teamsWithChosenChannelsStr =
+    chosenChannels && chosenChannels.item && chosenChannels.item.body && chosenChannels.item.body.toString()
+  const teamsWithChosenChannels = teamsWithChosenChannelsStr
```

Comment:
seems better to do the JSON parsing here, if only so we have reducers that can't throw exceptions, and better action typing